src/OutputWriters/netcdf_output_writer.jl

```diff
@@ -260,6 +260,11 @@ function NetCDFOutputWriter(model, outputs; filepath, schedule,
                                   compression = 0,
                                       verbose = false)
 
+    # We need to convert to a Dict with String keys if user provides a named tuple.
+    if outputs isa NamedTuple
+        outputs = Dict(string(k) => v for (k, v) in zip(keys(outputs), values(outputs)))
+    end
```

Comment:
Remove the if statement and replace with

```julia
outputs = dict_outputs(outputs)
```

along with, some where else, the function definitions

```julia
dict_outputs(outputs::Dict{<:String}) = outputs
dict_outputs(outputs) = Dict(string(k) => v for (k, v) in zip(keys(outputs), values(outputs)))
```

This design is more general, since it will convert any object that has `keys` and `values` to a `Dict{String}`, including named tuples, Dict's with symbols, and other objects that you might not be able to anticipate right now. This is the core advantage of multiple dispatch versus if statements.

Reply:
Ah good point. I usually use an `if` statement if I only have one thing to dispatch on but would be good to support any object that implements `keys` and `values`.